### ISSUE TITLE
Support GAE Flex environment in Core::Environment

### ISSUE THREAD

**Issue by quartzmo** [GoogleCloudPlatform/google-cloud-ruby#1075]
`Google::Cloud::Core::Environment` should support configuration for the new [GAE Flex environment](https://cloud.google.com/appengine/docs/flexible/).

**Comment by quartzmo** [comment]
After this issue is resolved, a follow-up PR to #1034 should change the configuration in the 3 `app.yaml.example` files from to `env: flex`, as [commented](https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1034/files#r87913643).

**Comment by hxiong388** [comment]
#1036 already exists for this problem. I have a PR ready to submit once #1034 is merged. :)

**Comment by quartzmo** [comment]
Thanks, I should have searched. Closing as dup.
